tests/x509/test_x509.py

```diff
@@ -63,6 +65,47 @@ def _load_cert(filename, loader, backend):
     return cert
 
 
+def _parse_cert(der):
+    # See the Certificate structured, defined in RFC 5280.
+    cert = DERReader(der).read_single_element(SEQUENCE)
+    tbs_cert = cert.read_element(SEQUENCE)
+    # Skip outer signature algorithm
+    _ = cert.read_element(SEQUENCE)
+    # Skip signature
+    _ = cert.read_element(BIT_STRING)
+    cert.check_empty()
+
+    # Skip version
+    _ = tbs_cert.read_optional_element(CONTEXT_SPECIFIC | CONSTRUCTED | 0)
+    # Skip serialNumber
+    _ = tbs_cert.read_element(INTEGER)
+    # Skip inner signature algorithm
+    _ = tbs_cert.read_element(SEQUENCE)
+    issuer = tbs_cert.read_element(SEQUENCE)
+    validity = tbs_cert.read_element(SEQUENCE)
+    subject = tbs_cert.read_element(SEQUENCE)
+    # Skip subjectPublicKeyInfo
+    _ = tbs_cert.read_element(SEQUENCE)
+    # Skip issuerUniqueID
+    _ = tbs_cert.read_optional_element(CONTEXT_SPECIFIC | CONSTRUCTED | 1)
+    # Skip subjectUniqueID
+    _ = tbs_cert.read_optional_element(CONTEXT_SPECIFIC | CONSTRUCTED | 2)
+    # Skip extensions
+    _ = tbs_cert.read_optional_element(CONTEXT_SPECIFIC | CONSTRUCTED | 3)
+    tbs_cert.check_empty()
+
+    not_before_tag, _ = validity.read_any_element()
+    not_after_tag, _ = validity.read_any_element()
+    validity.check_empty()
+
+    return {
```

Comment:
Can we make a named tuple or something for this structure. Using dicts to represent objects is a big anti-pattern in my view (and why tons of Python code is slow).

Reply:
Done.